UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -94,5 +96,113 @@ private async Task<CreateBatchResponse> CreateBatchResponse(HttpResponseMessage
+            SearchBatchResponse internalSearchBatchResponse = new SearchBatchResponse();
+            actualSearchBatchResponse.Entries = new List<BatchDetail>();
+            internalSearchBatchResponse.Entries = new List<BatchDetail>();
+            var accessToken = await authTokenProvider.GetManagedIdentityAuthAsync(fileShareServiceConfig.Value.ResourceId);
+            var productWithAttributes = GenerateQueryForFss(products);
+            var uri = $"/batch?limit={fileShareServiceConfig.Value.Limit}&start={fileShareServiceConfig.Value.Start}&$filter={fileShareServiceConfig.Value.ProductCode} {productWithAttributes}";
+
+            HttpResponseMessage httpResponse;
+
+            string payloadJson = string.Empty;
+            var productList = new List<string>();
+            var prodCount = products.Select(a => a.UpdateNumbers).Sum(a => a.Count);
+            do
+            {
+                httpResponse = await fileShareServiceClient.CallFileShareServiceApi(HttpMethod.Get, payloadJson, accessToken, uri);
+
+                if (httpResponse.IsSuccessStatusCode)
+                {
+                    SearchBatchResponse searchBatchResponse = await SearchBatchResponse(httpResponse);
+                    actualSearchBatchResponse.Count = searchBatchResponse.Count;
+                    foreach (var productItem in products)
+                    {
+                        foreach (var item in searchBatchResponse.Entries)
+                        {
+                            if (CheckProductDoesExistInResponseItem(item, productItem) && CheckEditionNumberDoesExistInResponseItem(item, productItem)
+                                && CheckUpdateNumberDoesExistInResponseItem(item, productItem))
+                            {
+                                var matchProduct = item.Attributes.Where(a => a.Key == "UpdateNumber");
+                                var updateNumber = matchProduct.Select(a => a.Value).FirstOrDefault();
+                                var compareProducts = $"{productItem.ProductName}|{productItem.EditionNumber}|{updateNumber}";
+                                if (!productList.Contains(compareProducts))
+                                {
+                                    internalSearchBatchResponse.Entries.Add(item);
+                                    productList.Add(compareProducts);
+                                }
+                            }
+                        }
+                    }
+                    uri = searchBatchResponse.Links.Next?.Href;
+                }
+                else
+                {
+                    logger.LogInformation(EventIds.QueryFileShareServiceNonOkResponse.ToEventId(), "File share service with uri {RequestUri} and responded with {StatusCode}", httpResponse.RequestMessage.RequestUri, httpResponse.StatusCode);
+                }
+            } while (httpResponse.IsSuccessStatusCode && internalSearchBatchResponse.Entries.Count < prodCount);
```

Comment:
As discussed in afternoon , can we add entries.count !=0 check in while loop.

Reply:
did changes accordingly